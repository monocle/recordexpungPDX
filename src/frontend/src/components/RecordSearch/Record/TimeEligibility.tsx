import React from "react";
import { TimeEligibilityData } from "./types";
import EligibilityReasons from "./EligibilityReasons";

interface Props {
  time_eligibility: TimeEligibilityData;
}

export default function TimeEligibility({ time_eligibility }: Props) {
  return <EligibilityReasons kind="time" eligibility={time_eligibility} />;
}
