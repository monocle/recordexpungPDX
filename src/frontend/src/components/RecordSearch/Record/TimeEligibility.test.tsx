import React from "react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { TimeEligibilityData } from "./types";
import TimeEligibility from "./TimeEligibility";

const timeEligibility = {
  status: "",
  reason: "a good reason",
  date_will_be_eligible: "a long time",
};

function doRender(time_eligibility: TimeEligibilityData) {
  return renderer
    .create(<TimeEligibility time_eligibility={time_eligibility} />)
    .toJSON();
}

it("renders correctly for an 'Eligible' status", () => {
  const time_eligibility = { ...timeEligibility, status: "Eligible" };
  expect(doRender(time_eligibility)).toMatchSnapshot();
});

it("renders correctly for a non specific status", () => {
  const time_eligibility = { ...timeEligibility, status: "not a real status" };
  expect(doRender(time_eligibility)).toMatchSnapshot();
});
