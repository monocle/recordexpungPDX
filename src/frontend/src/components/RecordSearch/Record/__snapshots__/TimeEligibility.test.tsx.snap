// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a non specific status 1`] = `
<div
  className="relative mb3 connect connect-time"
>
  <i
    aria-hidden="true"
    className="absolute fas fa-clock dark-blue bg-white outline-2-white z-1"
  />
  <div
    className="ml3 pl1"
  >
    <div
      className=""
    >
      <span
        className="fw7"
      >
        Time: 
      </span>
      a good reason
    </div>
  </div>
</div>
`;

exports[`renders correctly for an 'Eligible' status 1`] = `
<div
  className="relative mb3 connect connect-time"
>
  <i
    aria-hidden="true"
    className="absolute fas fa-check-circle green bg-white outline-2-white z-1"
  />
  <div
    className="ml3 pl1"
  >
    <span
      className="fw7"
    >
      Time:
    </span>
     Eligible Now
  </div>
</div>
`;
